EsentInterop/SetColumnHelpers.cs

```diff
@@ -149,14 +149,29 @@ public static void SetColumn(JET_SESID sesid, JET_TABLEID tableid, JET_COLUMNID
         /// <param name="data">The data to set.</param>
         /// <param name="grbit">SetColumn options.</param>
         public static void SetColumn(JET_SESID sesid, JET_TABLEID tableid, JET_COLUMNID columnid, byte[] data, SetColumnGrbit grbit)
+        {
+            int dataLength = (null == data) ? 0 : data.Length;
+            SetColumn(sesid, tableid, columnid, data, dataLength, grbit);
+        }
+        
+        /// <summary>
+        /// Modifies a single column value in a modified record to be inserted or to
+        /// update the current record.
+        /// </summary>
+        /// <param name="sesid">The session to use.</param>
+        /// <param name="tableid">The cursor to update. An update should be prepared.</param>
+        /// <param name="columnid">The columnid to set.</param>
+        /// <param name="data">The data to set.</param>
+        /// <param name="dataSize">The size of data to set.</param>
+        /// <param name="grbit">SetColumn options.</param>
+        public static void SetColumn(JET_SESID sesid, JET_TABLEID tableid, JET_COLUMNID columnid, byte[] data, int dataSize, SetColumnGrbit grbit)
         {

```

Comment:
Raise error if data is null but dataSize is > 0.

Reply:
Also, if dataSize is > data.Length

Reply:
Thanks! Fixid.